test/cast_test.exs

```diff
@@ -251,6 +251,37 @@ defmodule OpenApiSpec.CastTest do
 
       assert {:ok, %{data: "default"}} == cast(value: %{}, schema: schema)
     end
+
+    test "required fields respect read_write_scope" do
+      schema = %Schema{
+        type: :object,
+        required: [:data],
+        properties: %{
+          data: %Schema{
+            type: :string,
+            readOnly: true
+          }
+        }
+      }
+
+      assert {:ok, %{}} == cast(value: %{}, schema: schema, read_write_scope: :write)
+      assert {:error, errors} = cast(value: %{}, schema: schema, read_write_scope: :read)
+      assert {:error, ^errors} = cast(value: %{}, schema: schema)
+
+      assert [error] = errors
+      assert %Error{} = error
+      assert error.reason == :missing_field
+      assert error.path == [:data]
+      assert Error.message_with_path(error) == "#/data: Missing field: data"
+
+      assert {:ok, data} =
```

Comment:
Should this one be a validation error? A `readOnly` field is being supplied in a `:write` operation?
Can be a separate issue/pr if so.

Reply:
I initially had a fix for this in this PR, but ultimately decided against it.

Reason being is that I'm not entirely sure if it actually must be an error. OpenAPI spec version 3.0.3 [states](https://swagger.io/specification/#schemaReadOnly) that `readOnly` fields SHOULD NOT be part of a `write` operation, while version `3.1`, if I understood correctly, [now uses](https://spec.openapis.org/oas/latest.html#properties) definitions provided by Json schema draft 2020-12, which states that `readOnly` values "[MAY be ignored if sent to the owning authority, or MAY result in an error, at the authority's discretion](https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-02#section-9.4)". Following from that, I thought this change might be unnecessarily breaking things.

Reply:
👍 Good catch.